src/openqaoa-core/algorithms/qaoa/qaoa_workflow.py

```diff
@@ -303,6 +307,74 @@ def optimize(self, verbose=False):
         if verbose:
             print(f"optimization completed.")
         return
+    
+    def evaluate_circuit(
+        self, params: Union[List[float], Dict[str, List[float]], None] = None
+    ):
+        """
+        A method to evaluate the QAOA circuit at a given set of parameters
+
+        Parameters
+        ----------
+        params: list or dict or None
+            List of parameters or dictionary of parameters. Which will be used to evaluate the QAOA circuit.
+            If None, the variational parameters of the QAOA object will be used.
+
+        Returns
+        -------
+        result: dict
+            A dictionary containing the results of the evaluation:
+            - "expectation": the expectation value of the cost Hamiltonian
+            - "uncertainty": the uncertainty of the expectation value of the cost Hamiltonian
+            - "state": the state of the QAOA circuit output (if the QAOA circuit is evaluated on a state simulator)
+            - "counts": the counts of the QAOA circuit output (if the QAOA circuit is evaluated on a QPU or shot-based simulator)
+        """
+
+        # before evaluating the circuit we check that the QAOA object has been compiled
+        if self.compiled == False: raise ValueError("Please compile the QAOA before optimizing it!")
+
+        
+        ## Check the type of the input parameters and save them as a QAOAVariationalBaseParams object at the variable `params_obj`
+
+        # if no parameters are passed, we use the optimized parameters of the QAOA object
+        if params is None:
+            assert self.result != None, "No parameters have been passed and the QAOA object has not been optimized yet"
```

Comment:
We are not performing optimization here, so shouldn't we change the error message?

Reply:
changed, now the input must be list or dict